packages/ui-components/src/components/text-field/text-field.tsx

```diff
@@ -162,13 +169,17 @@ export class KvTextField implements ITextField, ITextFieldEvents {
 	}
 
 	private getType(): string {
-		return this.type === EInputFieldType.Number ? EInputFieldType.Text : this.type;
+		return this.getInputMaskEnabled() ? EInputFieldType.Text : this.type;
 	}
 
 	private getTooltipConfig = (): Partial<ITooltip> => {
 		return merge({}, DEFAULT_TEXT_TOOLTIP_CONFIG, this.tooltipConfig ?? {});
 	};
 
+	private getInputMaskEnabled = (): boolean => {
+		return this.type === EInputFieldType.Number && this.useInputMask;
```

Comment:
That way the `inputMaskRegex` can't allow characters only number. Imagine I would like to have an input for Portuguese plates (AA-00-AA)

Reply:
Can do now 👍